aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2353,6 +2353,70 @@ Tensor& linalg_vector_norm_out(const Tensor& self, const optional<Scalar>& opt_o
   return at::native::linalg_vector_norm_impl(self, opt_ord, opt_dim, keepdim, opt_dtype, result);
 }
 
+namespace {
+
+// Only performs checks not performed by linalg.norm
+void check_linalg_matrix_norm_args(
+    const Tensor& self,
+    IntArrayRef dim,
+    optional<ScalarType> dtype) {
+  TORCH_CHECK(
+      self.ndimension() >= 2,
+      "linalg.matrix_norm(): input tensor must be a matrix or batch of matrices");
+  ScalarType in_dtype = dtype.value_or(self.scalar_type());
+  TORCH_CHECK(
+      in_dtype == kFloat || in_dtype == kDouble || in_dtype == kComplexFloat ||
+          in_dtype == kComplexDouble,
+      "linalg.matrix_norm(): only supports floating point and complex dtypes, but got: ",
```

Comment:
nit: "only supports the float, double, cfloat and cdouble dtypes..."

because float16 and bfloat16 are also floating point dtypes, but they are not supported